Code/Core/FileIO/FileIO.cpp

```diff
@@ -375,10 +375,22 @@
 {
     #if defined( __WINDOWS__ )
         char buffer[ MAX_PATH ];
-        DWORD len = GetTempPath( MAX_PATH, buffer );
-        if ( len != 0 )
+        DWORD res = GetEnvironmentVariable("FASTBUILD_TEMP_PATH",
+                buffer,MAX_PATH);
+        if ( res == 0 )
+        {
+            DWORD len = GetTempPath( MAX_PATH, buffer );
+            if ( len != 0 )
+            {
+                output = buffer;
+                output += '\\';
+                return true;
+            }
+        }
+        else
         {
             output = buffer;
+            output += '\\';
```

Comment:
Seems like this would add a trailing backslash even if the environment variable already has one. Also, according to the MSDN documentattion, the ::GetTempPath call should return the path with a trailing backslash already (though I've seen evidence that this might not be true).  Seems like the best bet would be to check if it's already there and add it if not in both cases.